contracts/defi/provider/MakerProvider.sol

```diff
@@ -0,0 +1,818 @@
+    function bid(uint wad) public view returns (uint);
+
+    function open() external returns (bytes32 cup);
+    function join(uint wad) external; // Join PETH
+    function exit(uint wad) external; // Exit PETH
+    function give(bytes32 cup, address guy) external;
+    function lock(bytes32 cup, uint wad) external;
+    function free(bytes32 cup, uint wad) external;
+    function draw(bytes32 cup, uint wad) external;
+    function wipe(bytes32 cup, uint wad) external;
+    function shut(bytes32 cup) external;
+    function bite(bytes32 cup) external;
+}
+
+interface IMakerVox {
+    function par() external returns (uint);
+}
+
+interface IDSValue {
+    function peek() external view returns (bytes32, bool);
+    function read() external view returns (bytes32);
+    function poke(bytes32 wut) external;
+    function void() external;
+} 
+
+/**
+ * @title MakerProvider
+ * @dev Wrapper contract to integrate MakerDao.
+ * The first item of the oracles array is the Maker Tub contract, the second is the Uniswap Factory 
+ * that may be used to orchestrate the exchange of tokens.
+ * @author Olivier VDB - <olivier@argent.xyz>, Julien Niset - <julien@argent.xyz>
+ */
+contract MakerProvider is Loan { //}, Leverage {
+
+    // Mock token address for ETH
+    address constant internal ETH_TOKEN_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    // Method signatures to reduce gas cost at depoyment
+    bytes4 constant internal CDP_DRAW = bytes4(keccak256("draw(bytes32,uint256)"));
+    bytes4 constant internal CDP_WIPE = bytes4(keccak256("wipe(bytes32,uint256)"));
+    bytes4 constant internal CDP_SHUT = bytes4(keccak256("shut(bytes32)"));
+    bytes4 constant internal CDP_JOIN = bytes4(keccak256("join(uint256)"));
+    bytes4 constant internal CDP_LOCK = bytes4(keccak256("lock(bytes32,uint256)"));
+    bytes4 constant internal CDP_FREE = bytes4(keccak256("free(bytes32,uint256)"));
+    bytes4 constant internal CDP_EXIT = bytes4(keccak256("exit(uint256)"));
+    bytes4 constant internal WETH_DEPOSIT = bytes4(keccak256("deposit"));
+    bytes4 constant internal WETH_WITHDRAW = bytes4(keccak256("withdraw(uint256)"));
+    bytes4 constant internal ERC20_APPROVE = bytes4(keccak256("approve(address,uint256)"));
+    bytes4 constant internal ETH_TOKEN_SWAP_OUTPUT = bytes4(keccak256("ethToTokenSwapOutput(uint256,uint256)"));
+    bytes4 constant internal ETH_TOKEN_SWAP_INPUT = bytes4(keccak256("ethToTokenSwapInput(uint256,uint256)"));
+    bytes4 constant internal TOKEN_ETH_SWAP_INPUT = bytes4(keccak256("tokenToEthSwapInput(uint256,uint256,uint256)"));
+
+    using SafeMath for uint256;
+
+    /* *************** Events *************************** */
+
+    event CdpOpened(address indexed wallet, bytes32 cup, uint256 pethCollateral, uint256 daiDebt);    
+    event CdpUpdated(address indexed wallet, bytes32 cup, uint256 pethCollateral, uint256 daiDebt);    
+    event CdpClosed(address indexed wallet, bytes32 cup);
```

Comment:
I think we can remove those events now that we have `LoanOpened`, `LoanClosed`, `CollateralAdded`, etc in `LoanManager`. I confirmed with Matt that the client does not need them.

Reply:
Agreed and done.